precompile/contracts/feemanager/contract_test.go

```diff
@@ -0,0 +1,273 @@
+			assertState: func(t *testing.T, state *state.StateDB) {
+				feeConfig := GetStoredFeeConfig(state)
+				lastChangedAt := GetFeeConfigLastChangedAt(state)
+				require.Equal(t, testFeeConfig, feeConfig)
+				require.EqualValues(t, big.NewInt(6), lastChangedAt)
+			},
+		},
+		"get initial fee config": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				return PackGetFeeConfigInput()
+			},
+			suppliedGas: GetFeeConfigGasCost,
+			config: &Config{
+				InitialFeeConfig: &testFeeConfig,
+			},
+			readOnly: true,
+			expectedRes: func() []byte {
+				res, err := PackFeeConfig(testFeeConfig)
+				require.NoError(t, err)
+				return res
+			}(),
+			assertState: func(t *testing.T, state *state.StateDB) {
+				feeConfig := GetStoredFeeConfig(state)
+				lastChangedAt := GetFeeConfigLastChangedAt(state)
+				require.Equal(t, testFeeConfig, feeConfig)
+				require.EqualValues(t, testBlockNumber, lastChangedAt)
+			},
+		},
+		"get last changed at from non-enabled address": {
+			caller: noRoleAddr,
+			preCondition: func(t *testing.T, state *state.StateDB) {
+				err := StoreFeeConfig(state, testFeeConfig, contract.NewMockBlockContext(testBlockNumber, 0))
+				require.NoError(t, err)
+			},
+			input: func() []byte {
+				return PackGetLastChangedAtInput()
+			},
+			suppliedGas: GetLastChangedAtGasCost,
+			readOnly:    true,
+			expectedRes: common.BigToHash(testBlockNumber).Bytes(),
+			assertState: func(t *testing.T, state *state.StateDB) {
+				feeConfig := GetStoredFeeConfig(state)
+				lastChangedAt := GetFeeConfigLastChangedAt(state)
+				require.Equal(t, testFeeConfig, feeConfig)
+				require.Equal(t, testBlockNumber, lastChangedAt)
+			},
+		},
+		"readOnly setFeeConfig with noRole fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := PackSetFeeConfig(testFeeConfig)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: SetFeeConfigGasCost,
+			readOnly:    true,
+			expectedErr: vmerrs.ErrWriteProtection.Error(),
+		},
+		"readOnly setFeeConfig with allow role fails": {
+			caller: enabledAddr,
+			input: func() []byte {
+				input, err := PackSetFeeConfig(testFeeConfig)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: SetFeeConfigGasCost,
+			readOnly:    true,
+			expectedErr: vmerrs.ErrWriteProtection.Error(),
+		},
+		"readOnly setFeeConfig with admin role fails": {
+			caller: adminAddr,
+			input: func() []byte {
+				input, err := PackSetFeeConfig(testFeeConfig)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: SetFeeConfigGasCost,
+			readOnly:    true,
+			expectedErr: vmerrs.ErrWriteProtection.Error(),
+		},
+		"insufficient gas setFeeConfig from admin": {
+			caller: adminAddr,
+			input: func() []byte {
+				input, err := PackSetFeeConfig(testFeeConfig)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: SetFeeConfigGasCost - 1,
+			readOnly:    false,
+			expectedErr: vmerrs.ErrOutOfGas.Error(),
+		},
```

Comment:
Note: this drops two allow list tests from the original.

We should see if we can turn the allow list test suite into a a test suite that can be imported and used by all of the precompiles that use the allow list.

Reply:
created #497

Reply:
My bad, I forgot you already created this issue. Thanks for being on top of this